op-challenger/game/fault/trace/outputs/split_test.go

```diff
@@ -0,0 +1,325 @@
+			localClaimPos := types.NewPositionFromGIndex(big.NewInt(1))
+			requireSameValue(t, provider, claimPos, asBottomTraceProvider(t, provider).AlphabetTraceProvider, localClaimPos)
+			requireSameValue(t, provider, claimPos.Attack(), asBottomTraceProvider(t, provider).AlphabetTraceProvider, localClaimPos.Attack())
+			requireSameValue(t, provider, claimPos.Attack().Defend(), asBottomTraceProvider(t, provider).AlphabetTraceProvider, localClaimPos.Attack().Defend())
+		})
+	}
+}
+
+func requireSameValue(t *testing.T, a types.TraceProvider, aPos types.Position, b types.TraceProvider, bPos types.Position) {
+	// Check Get returns the same results
+	aValue, err := a.Get(context.Background(), aPos)
+	require.NoError(t, err)
+	bValue, err := b.Get(context.Background(), bPos)
+	require.NoError(t, err)
+	require.Equal(t, aValue, bValue)
+
+	// Check GetStepData returns the same results
+	aPrestate, aProofData, aPreimageData, err := a.GetStepData(context.Background(), aPos)
+	require.NoError(t, err)
+	bPrestate, bProofData, bPreimageData, err := b.GetStepData(context.Background(), bPos)
+	require.NoError(t, err)
+	require.Equal(t, aPrestate, bPrestate)
+	require.Equal(t, aProofData, bProofData)
+	require.Equal(t, aPreimageData, bPreimageData)
+}
+
+func TestBottomProviderAttackingTopLeaf(t *testing.T) {
+	tests := []struct {
+		name  string
+		setup func(t *testing.T, gameBuilder *test.GameBuilder) (ref types.Claim, pos types.Position, expectPre types.Claim, expectPost types.Claim)
+	}{
+		// There are 4 leaf nodes that can be accessed in the top tree of depth 3: 8, 10, 12, 14
+		// Then you can attack and defend any of those to challenge all blocks
+		// We can then use these setups to test any other reference claim descending from what these setup since
+		// that whole subtree should have the same pre and post claim from the top provider.
+		{"attackTopLeafGIndex8", attackTopLeafGIndex8},
+		{"defendTopLeafGIndex8", defendTopLeafGIndex8},
+		{"attackTopLeafGIndex10", attackTopLeafGIndex10},
+		{"defendTopLeafGIndex10", defendTopLeafGIndex10},
+		{"attackTopLeafGIndex12", attackTopLeafGIndex12},
+		{"defendTopLeafGIndex12", defendTopLeafGIndex12},
+		{"attackTopLeafGIndex14", attackTopLeafGIndex14},
+		{"attackTopLeafGIndex14", defendTopLeafGIndex14},
+	}
+	for _, tCase := range tests {
+		tCase := tCase
+		t.Run(tCase.name, func(t *testing.T) {
+			_, selector, gameBuilder := setupAlphabetSplitSelector(t)
+
+			ref, pos, expectedPre, expectedPost := tCase.setup(t, gameBuilder)
+
+			runTest := func(ref types.Claim, pos types.Position) {
+				t.Run(fmt.Sprintf("Ref-d%vi%v_Pos-d%vi%v", ref.Depth(), ref.IndexAtDepth(), pos.Depth(), pos.IndexAtDepth()), func(t *testing.T) {
+					provider, err := selector(context.Background(), gameBuilder.Game, ref, pos)
+					require.NoError(t, err)
+					requireBottomProviderForClaims(t, provider, expectedPre, expectedPost)
+				})
+			}
+
+			// Check we get the same pre and post for any reference claim lower in the game
```

Comment:
We wind up attacking and defending both in terms of the position relative to the reference claim and in how we add reference claims to test all combinations.  The only restriction is that for the first level, whether the reference claim attacks or defends the top game leaf changes which pre and post state you should expect, so we have the `attackTopLeafGIndex` and `defendTopLeafGIndex` variants which have the top leaf as the reference claim and the position either attacking or defending that. So then we need to ensure that when we generate new reference claims that we only generate one that matches the attack/defend of the position - otherwise we'd change the pre/post expectations and the test would fail.